etcd/etcd.go

```diff
@@ -2,20 +2,31 @@ package etcd
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"github.com/weldpua2008/suprasched/core"
 	"go.etcd.io/etcd/client/v3"
 	"log"
-	"reflect"
-	"strconv"
+	"strings"
 	"time"
 )
 
+type UniversalObject interface {
+	GetObjId() string
+}
+
 var (
-	dialTimeout    = 2 * time.Second
+	dialTimeout    = 5 * time.Second
 	requestTimeout = 10 * time.Second
+	tempObj        UniversalObject
 )
 
-func GetKV(key string, endpoint string) map[string]string { // do "defer client.Close()" and "defer cancel()"
+func PutKV_v2(obj UniversalObject, key string, endpoint string) {
+
+	obj_type := CheckStructType(obj)  // Checking the type of struct we received
+	json_data, _ := json.Marshal(obj) // Converts the object to a json file as one long string
```

Comment:
Had you tried to implement something like?:
```go
if obj.GetKind() == 'job'. 
```
where `GetKind()`. will return value of Kind

Reply:
I have another function that does this: CheckGetStructType & CheckStructType